docs/25_git_stash.md

```diff
@@ -0,0 +1,60 @@
+## Git Stash
+
+Git stash takes the changes you have made to your working tree ("dirty") and puts them in a holding pattern until you are ready to re-apply them so you have a fresh working directory ("clean").
+
+Common use cases for using git stash are
+
+- Needing to change focus from the task you are currently working on to a new task
+- Forgetting to switch working branches
+- Needing to apply changes to multiple branches
+
+When these scenarios happen, figuring out what to do with your in-progress work might be difficult. Do you commit it even though it isn't "ready to be committed"? Do you create a new branch with your changes and come back to it? Do you need to delete your changes and just re-create the work when you can come back to this task? Do you manually copy-and-paste the changes across branches?
+
+There are different solutions to these problems, but this section is going to review the use of `git stash` to store changes you have made in the event that you do not want to commit them to the project's history.
+
+### How to use `git stash`
+
+The `git stash` command enables you to store changes that live in the working directory (unstaged changes) and staging area (staged changes) for later use. These stored changes, like the `reflog`, are only reflected on your *local copy* of the repository. The latest stash is stored in `refs/stash` and older stashes are found in the `reflog` of that reference.
+
+You can have multiple stashes, and they can be referenced by specifying the stash index or they can be named using `reflog` syntax (example: `stash@{2.hours.ago}`). You can also add a descriptive message to stashes with `git stash save "message"`. Once you are ready to retrieve those changes, you can apply and delete them with `git pop` or apply and continue to store them with `git stash apply`.
+
+#### Commands
+
+- `git stash`
+  - Use this when you want to record the current working state, but want a clean working directory
+  - This will "stash" your local modifications and revert the working directory to match the `HEAD` commit
+  - `git stash push`
+    - Saves local changes to the stash and then rolls back to `HEAD` in the working tree and index
+      - `git stash push -m <message>` gives the message along with the stashed state
+      - Options include `-m|--message <message>`, `-p|--patch`, `-`
```

Comment:
```suggestion
      - Options include `-m|--message <message>`, `-p|--patch`, `--`
```

I’m not sure if this was your intention or not.

Reply:
Good catch.